WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsFactory.swift

```diff
@@ -0,0 +1,13 @@
+import UIKit
+
+struct SettingsFactory {
+    /// Creates a Settings view controller
+    ///
+    static func settings() -> UIViewController {
+        let viewModel = SettingsViewModel(stores: ServiceLocator.stores,
+                                          storageManager: ServiceLocator.storageManager)
+        let settingsViewController = SettingsViewController(viewModel: viewModel)
+        viewModel.presenter = settingsViewController
+        return settingsViewController
+    }
+}
```

Comment:
Thanks a lot for reviewing @itsmeichigo 

My thought process was to hide the initialization (DI) and setup (setting presenter for `ViewModel`) details inside a factory. I am hoping that we will have to inject more types such as FeatureFlag and etc in future.

What do you think?

Reply:
Thanks for the explanation. My understanding of the factory pattern is that we need it to create different kinds of instances that conform a shared protocol. I'm concerned that using this here may be misleading, since we're not having different kinds of Settings in the app. 

For DI we can set default values for `stores` and `storageManager` to simplify the SettingsViewModel's initilizer. We can also set up the presenter for the view model when initilizing the view controller. That way we can tidy up the setup - how does that sound?

Reply:
You are right @itsmeichigo. 
I made the necessary changes. Kindly take a look at `SettingsViewController` 's init method and share your thoughts.

Reply:
Good point! I think we can keep the view model a constant and still mutable by making `SettingsViewModelInput` conform to `AnyObject`. That'll work right?

Reply:
Great, that will work. Done in https://github.com/woocommerce/woocommerce-ios/pull/5393/commits/7d971ce68b09caafe40853c3477e6e545f210635.
Thank you for your inputs! 🙇‍♂️